### PR TITLE
DPRO-3407: Developer overview

### PR DESCRIPTION
**Pull request by rskonnord-plos**
First iteration of my general-purpose brain dump. It's targeted mostly at new PLOS developers but also serviceable for prospective volunteer developers and as a general reference.

Most of it is meant to be at an introductory level -- not simple necessarily, but the fundamental concepts you would need to understand to do stuff in the middle tier. There are a few places (particularly around the request mapping stuff) where I might have gotten a little too far in the weeds while trying to explain some of the hairier pieces of code. These aren't *necessarily* things you would have to understand to work on the code base day-to-day, but they are still things that (1) a dev will have to deal with every so often and (2) would more likely want a lifeline. Perhaps it would help to separate them from the higher-level topics in a new section toward the bottom.

When reviewing this, please consider:

1. Are there any areas where you would like to see more detail?
2. Are there any areas where I'm maybe explaining in too much detail, to the point of confusing new developers?
3. Are there any areas (classes, concepts, layers, components) that I've skipped over entirely and you would like to see me cover in a new section)?

### CODE REVIEW

**Comment by rskonnord-plos** [comment]
The formatted page can be seen at https://github.com/PLOS/ambraproject/blob/dev-DPRO-3407/Developer-Overview.md

File: Developer-Overview.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+In our Java code, the [`Site`][Site] and [`Theme`][Theme] classes represent
+these two abstractions.
+
+  [Site]: https://github.com/PLOS/wombat/blob/master/src/main/java/org/ambraproject/wombat/config/site/Site.java
+  [Theme]: https://github.com/PLOS/wombat/blob/master/src/main/java/org/ambraproject/wombat/config/theme/Theme.java
+
+Note the relationship between a **site** and a **journal**. A journal may be
+represented by more than one site, as in the "desktop and mobile" example
+above. To resolve a journal key into a link to another site, it is necessary to
```
Comment (sjtower) [inline Developer-Overview.md]: This part is a bit confusing. Can we point to an example in the Wombat code?

**Comment by sjtower** [comment]
:+1:

File: Developer-Overview.md (hunk 1 above)
Comment (rskonnord-plos) [inline Developer-Overview.md]: @sjtower How is this? 26b04856

**Pull request merged**